projects/vdk-plugins/vdk-structlog/tests/test_plugin.py

```diff
@@ -1,33 +1,120 @@
-        # and mock large parts of it - e.g passed our own plugins
-        runner = CliEntryBasedTestRunner(structlog_plugin)
-
-        result: Result = runner.invoke(
-            ["run", jobs_path_from_caller_directory("job-using-a-plugin")]
-        )
-        cli_assert_equal(0, result)
+JOB_NAME = "job-with-bound-logger"
+BOUND_TEST_KEY = "bound_test_key"
+BOUND_TEST_VALUE = "bound_test_value"
+EXTRA_TEST_KEY = "extra_test_key"
+EXTRA_TEST_VALUE = "extra_test_value"
+EXCLUDED_VALUE = "excluded_value"
+
+
+STOCK_FIELDS = ["level", "file_name", "line_number", "vdk_job_name"]
+STOCK_FIELD_REPRESENTATIONS = {
+    "level": "INFO",
+    "file_name": "10_dummy.py",
+    "line_number": ":23",
+    "vdk_job_name": JOB_NAME,
+}
+
+
+def test_structlog_console():
+    for log_format in ["console", "json"]:
+        with mock.patch.dict(
+            os.environ,
+            {
+                "VDK_LOGGING_METADATA": f"timestamp,level,file_name,line_number,vdk_job_name,{BOUND_TEST_KEY},{EXTRA_TEST_KEY}",
+                "VDK_LOGGING_FORMAT": log_format,
+            },
+        ):
+            runner = CliEntryBasedTestRunner(structlog_plugin)
+
+            result: Result = runner.invoke(
+                ["run", jobs_path_from_caller_directory("job-with-bound-logger")]
+            )
+
+            logs = result.output.split("\n")
+
+            log_with_no_bound_context = [
+                x for x in logs if "Log statement with no bound context" in x
+            ][0]
+            log_with_bound_context = [
+                x for x in logs if "Log statement with bound context" in x
+            ][0]
+            log_with_bound_and_extra_context = [
+                x
+                for x in logs
+                if "Log statement with bound context and extra context" in x
+            ][0]
+
+            _assert_cases(
+                log_with_no_bound_context,
+                log_with_bound_context,
+                log_with_bound_and_extra_context,
+            )
+
+        for removed_field in STOCK_FIELDS:
+            shown_fields = [field for field in STOCK_FIELDS if field != removed_field]
+
+            with mock.patch.dict(
+                os.environ,
+                {
+                    "VDK_LOGGING_METADATA": ",".join(shown_fields)
+                    + "bound_test_key,extra_test_key"
+                },
+            ):
+                runner = CliEntryBasedTestRunner(structlog_plugin)
+
+                result: Result = runner.invoke(
+                    ["run", jobs_path_from_caller_directory("job-with-bound-logger")]
+                )
+
+                test_log = [
+                    x
+                    for x in result.output.split("\n")
+                    if "Log statement with bound context and extra context" in x
+                ][0]
+
+                # check that the removed_field in not shown in the log
+                assert STOCK_FIELD_REPRESENTATIONS[removed_field] not in test_log
```

Comment:
I think you should also assert that the shown fields are indeed shown. Otherwise, it can get messy. For example, I know that the line number is not going to show up in the json logs like this `:23`. So just asserting that it's not there doesn't add anything to the test. We have to assert that the fields that weren't removed are present as well.

Reply:
included two sets of of field representations for console and json each